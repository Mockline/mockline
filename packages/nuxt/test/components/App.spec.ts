import { describe, it, expect } from 'vitest'
import type { TypeOf } from 'zod'
// @ts-ignore
import { MApp } from '#components'
import ComponentRender from '../component-render'

describe('App', () => {
  it.each(
    [
      ['transparent', {}],
      ['<MApp :tooltip="{ delayDuration: 5 }" class="mt-5" transparent>Slot Filled here</MApp>']
      // @ts-expect-error
    ])('renders %s correctly', async (nameOrHtml: string, options: TypeOf<typeof MApp.props>) => {
      const html = await ComponentRender(nameOrHtml, options, MApp)
      expect(html).toMatchSnapshot()
  })
})
