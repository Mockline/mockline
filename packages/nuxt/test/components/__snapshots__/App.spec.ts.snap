// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`App > renders <MApp :tooltip="{ delayDuration: 5 }" class="mt-5" transparent>Slot Filled here</MApp> correctly 1`] = `"<div class="bg-transparent mt-5">Slot Filled here</div>"`;

exports[`App > renders transparent correctly 1`] = `"<div class="bg-[var(--color-bg-primary)]"></div>"`;
